Do not replace another `added_to_cart` vanilla handler if there is one

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Do not replace another `added_to_cart` vanilla handler if there is one.

Or jQuery event to Vanilla listener hack for `added_to_cart` may remove any previous listeners if there are any.

For other events: `onupdated_wc_div` and `onremoved_from_cart` we prevent that.

### Checks:
<!-- Mark completed items with an [x] -->
* [x] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Simulate another extension listening to the event
```php
add_action(
	'wp_head',
	function () {
		wp_add_inline_script(
			'woocommerce-google-analytics-integration',
			'document.body.onadded_to_cart = () => { console.log("Other extension listens to added to cart"); };',
			'before',
		);
	},
);
```
3. Go to shop page
4. Click add to cart
5. You should see the console log, and event should be tracked in https://tagassistant.google.com/


### Additional details:

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Tweak - Do not replace other extensions' `add_to_cart` listeners.

tomalec committed:
Do not replace another `added_to_cart` vanilla handler if there is one

eason9487 commented on assets/js/src/integrations/classic.js:
```diff
@@ -40,15 +40,26 @@ export function classicTracking(
+		...rest
+	) => {
+		if ( typeof oldAddedToCart === 'function' ) {
+			oldAddedToCart( e, fragments, cartHash, button, ...rest );
```
If the goal is to minimize the impact on other extensions, it's also possible to carry `this` over correctly. For example, I would expect the following function to be called with [the `body` node](https://github.com/woocommerce/woocommerce/blob/8.9.3/plugins/woocommerce/client/legacy/js/frontend/add-to-cart.js#L116) as `this` instead of the `window` object, even though `this` isn't normally used.

```php
add_action(
	'wp_head',
	function () {
		wp_add_inline_script(
			'woocommerce-google-analytics-integration',
			'document.body.onadded_to_cart = function() { console.log( this, ...arguments ); };',
			'before',
		);
	},
);
```

eason9487 reviewed (commented):
Thanks for improving the compatibility with other extensions!

The code works well but I would like to know if it could consider carrying the `this` context as well.

tomalec committed:
Preserve `this` for Vanilla handlers to jQuery events

Address https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/438#pullrequestreview-2122048685

tomalec commented:
Thanks for catching that. :) Fixed in https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/438/commits/dd9211fa3df8a91209f7765722badaf18e65ca83. Would you mind another round @eason9487

eason9487 reviewed (approved):
Thanks for the further adjustments! LGTM.
